### ISSUE TITLE
PGN 130306 (Wind)

### ISSUE THREAD

**Issue by TwoCanPlugIn** [antipole2/VDR2#1]
Excellent utility.

Note however that in constructing the MWV sentence you are hardcoding the wind reference field 'R' meaning relative. Be aware that the 4th field of PGN 130306 is the wind reference. That should be taken into consideration.

**Comment by antipole2** [comment]
Thank you for this.  It is just the sort of feedback I need.

The 4th field of PGN 130306 is a value 0-4 with the following meanings
````
"EnumValues": [
		{
		        "name": "True (ground referenced to North)",
		        "value": 0
		},
		{
		       "name": "Magnetic (ground referenced to Magnetic North)",
		       "value": 1
		},
		{
			"name": "Apparent",
			"value": 2
		},
		{
			"name": "True (boat referenced)",
			"value": 3
		},
		{
			"name": "True (water referenced)",
			"value": 4
		}
	]
````
How do you advise this value is mapped to an MWV sentence?

**Comment by TwoCanPlugIn** [comment]
PGN 130306 Wind Reference  value of 2 (Apparent) maps to 'R' (relative) in the wind reference field of the MWV sentence.

The 'T' in the MWV sentence means Theoretical (wind speed and direction relative to the boat as if it was stationary), which might map to 3, (True Boat Referenced).  In reality, most wind instruments on a boat just transmit Apparent wind, it's usually an instrument or MFD that performs the True Wind calculations.

**Comment by antipole2** [comment]
I've updated the conversion to:
````
function convert130306(obj){	// wind
	var sentence;
	if (trace2k) print(JSON.stringify(obj, null, "\t"), "\n");
	angle = obj.windAngle * 57.29578;	// angle from radians to degrees
	speed = obj.windSpeed * 1.943844;	// speed from m/s to knots
	switch (obj.reference){
		case "Apparent":
			ref = "R"; break;
		case "True (boat referenced)":
			ref = "T"; break;
		default: throw("PGN130306 has unsupported reference " + obj.reference + " please report this");
		}
	sentence = "$" + sender +"MWV," +angle.toFixed(2) + "," + ref + "," + speed.toFixed(2) + ",K,A";
	buffer += sentence + "*" + NMEA0183checksum(sentence) + "\n";
	}
````

Thinking:  Maybe the decoder should set the enum value rather than its name? 
Or perhaps both?

**Comment by TwoCanPlugIn** [comment]
Whatever works for you and provides readability.

Unlike C/C++  and similar to Visual Basic and C# you have the benefit of not being confined to using numeric values in switch statements.

**Comment by antipole2** [comment]
Above change implemented in v1.1.
Have put issue of what to set on my to-do list.
